views/extras.tmpl

```diff
@@ -117,6 +117,131 @@
             toggleToLightClasses(event.data.newBody)
         }
     })
+
+
+    function updateConnectionStatus(msg, connected) {
+        var el = $('#connection');
+        el.removeClass('hidden')
+        if (connected) {
+            el.addClass('connected')
+            el.removeClass('disconnected')
+        } else {
+            el.removeClass('connected');
+            el.addClass('disconnected');
+        }
+        el.html(msg + '<div></div>')
+    }
+
+
+
+    function getSocketURI(loc) {
+        var protocol = (loc.protocol === 'https:') ? 'wss' : 'ws'
+        var route = (loc.pathname === "/") ? "/ws" : loc.pathname + '/ws'
+        return protocol + "://" + loc.host + route
+    }
+
+    var ws; // websocket global
+    function createWebSocket(loc) {
+        var uri = getSocketURI(loc)
+        ws = new MessageSocket(uri);
+        console.log("createdWebSocket", uri, ws)
+        ws.registerEvtHandler("open", function() {
+            console.log('Connected')
+            updateConnectionStatus('Connected', true);
+        });
+
+        ws.registerEvtHandler("close", function() {
+            console.log('Disconnected')
+            updateConnectionStatus('Disconnected', false);
+        });
+
+        ws.registerEvtHandler("error", function(evt) {
+            console.log("WebSocket error:", evt);
+            updateConnectionStatus('Disconnected', false);
+        });
+
+        ws.registerEvtHandler("ping", function(evt) {
+            console.log("ping. users online: ", evt)
+            ws.send("pong", "copy")
+        });
+        var updateBlockData = function (event) {
+            console.log("Received newblock message", event);
+            var newBlock = JSON.parse(event);
+            var confirms = $(".confirmations");
```

Comment:
If the data attributes with the confirmation heights are set on the elements holding the confirmations values themselves, the confirmation values can be updated in a single loop.

Check out the ages are updated  https://github.com/dcrdata/dcrdata/blob/master/views/extras.tmpl#L138

Should be possible to eliminate `initialBlocks` and `heights`

Reply:
Funny, I just noticed we could be passing unixTime in directly to humanize.timeSince in the ages updater, instead of calling `.data('age')` and second time. https://github.com/dcrdata/dcrdata/blob/master/views/extras.tmpl#L142